src/ast.rs

```diff
@@ -76,19 +78,36 @@ impl Union {
     /// be consumed by the JSON type. In a similar fashion, a table schema is determined
     /// to be nullable or required via occurances of null types in unions.
     pub fn collapse(&self) -> Tag {
-        let nullable: bool = self.items.iter().any(|x| x.is_null());
+        let is_null = self.items.iter().any(|x| x.is_null());
 
         if self.items.is_empty() {
             panic!("empty union is not allowed")
         } else if self.items.len() == 1 {
             return Tag {
                 name: None,
-                nullable: nullable,
+                nullable: is_null,
                 data_type: self.items[0].data_type.clone(),
             };
         }
 
-        let items: Vec<&Box<Tag>> = self.items.iter().filter(|x| !x.is_null()).collect();
+        let items: Vec<Box<Tag>> = self
```

Comment:
Sometimes you can leave out the inner type: `let items: Vec<_> = ...`.
It might work here, but you don't have to follow my suggestion if you prefer the most explicit version.

Reply:
I like the explicit version because it reminds me of the current type, but I'll keep in mind the syntax.